.github/workflows/main.yml

```diff
@@ -0,0 +1,83 @@
+# This is a basic workflow to help you get started with Actions
+
+name: CI
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+jobs:
+  job1:
+    name: Build linux
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, ubuntu-20.04]
+        python-version: [3.8, 3.x]
+        compiler_v: [7,8,9]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        sudo add-apt-repository ppa:ubuntu-toolchain-r/test  
+        sudo apt-get update
+        sudo apt-get install -y gcc-${{ matrix.compiler_v }} \
+                                g++-${{ matrix.compiler_v }}
+        python -m pip install --upgrade pip
+        python -m pip install pytest numpy
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
+        if [ "${{ matrix.compiler_v }}" == '7' ];then
+          pip install -e .[compression]
+        else
+          pip install -e .
+        fi
+      env:
+        PYTHONPATH: $PWD
+        CC: gcc-${{ matrix.compiler_v }}
+        CXX: g++-${{ matrix.compiler_v }}
+    - name: Lint with flake8
+      run: flake8 .
```

Comment:
Standard practice is for these things to be in separate workflows. See Devito's CI where flake is a separate workflow. Or am I misunderstanding something?

Reply:
you're right I wil do it in that way